design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,373 @@
+|----|-------------------------------------------------------------------------------------------|---------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------|---------------------------|------------|
+| 1  | authn-azure is not whitelisted in ENV or enabled in DB                                    | VM sends authentication request to Conjur with valid Azure token    | 500 Internal Server Error response with an error body with the relevant error message (AuthenticatorNotWhitelisted) is returned and logged   | Integration               |            |
+| 2  | Policy is loaded w/o "provider-uri" variable or value is missing                          | VM sends authentication request to Conjur with valid Azure token    | 500 Internal Server Error response with an error body with the relevant error message (RoleNotAuthorizedOnWebservice) is returned and logged | Integration               |            |
+| 3  | Policy is loaded w/o defining authenticator webservice                                    | VM sends authentication request to Conjur with valid Azure token    | 500 Internal Server Error response with an error body with the relevant error message (WebserviceNotFound) is returned and logged            | Integration               |            |
+| 4  | An incorrectly/undefined provider-uri (Conjur couldn't connect to provider)               | VM sends authentication request to Conjur with valid Azure token    | 500 Internal Server Error response with an error body with the relevant error message (ProviderDiscoveryTimeout) is returned and logged      | Integration               |            |
+| 5  | Conjur can make contact with provider but couldn't confirm token signature                | VM sends authentication request to Conjur with invalid Azure token  | 500 Internal Server Error response with an error body with the relevant error message (ProviderTokenInvalid) is returned and logged          | Integration               |            |
+| 6  | One or more annotations are missing from the host policy **                               | VM sends authentication request to Conjur with valid Azure token    | 500 Internal Server Error response with an error body with the relevant error message (MissingRequestParam) is returned and logged           | Integration               |            |
+| 7  | One or more fields in Azure token is missing                                              | VM sends authentication request to Conjur with invalid Azure token  | 500 Internal Server Error response with an error body with the relevant error message (TokenFieldNotFoundOrEmpty) is returned and logged     | Integration               |            |
```

Comment:
let's pinpoint this to the `xms_mirid` field as it's the only one that we are looking for.

Reply:
And update the actual log message for `TokenFieldNotFoundOrEmpty` to be 
"xms_mirid field not found or empty"?

Reply:
I am wary to be specific here b/c I imagine the future we will want to check other fields of tokens, not just `xms_mirid`

Reply:
I was referring to the `Given` section. The error should still be `TokenFieldNotFoundOrEmpty` as it's a field in the token that is missing, not the field `token`. Let's make sure that we log also the field that is missing.